### PR TITLE
Add variant for CheckBoxGroup with array of objects for options

### PR DESCRIPTION
**Pull request by taysea**
Adds a variant demonstrating that options can be an array of objects as opposed to just an array of strings.

### CODE REVIEW

File: aries-site/src/examples/components/checkboxgroup/CheckBoxGroupObjectExample.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    <Form>
+      <FormField
+        name="checkboxgroup-objectoptions"
+        fill
+        htmlFor="objectoptions-checkboxgroup"
+        label="Label"
+      >
+        <CheckBoxGroup
+          options={objectOptions}
+          name="checkboxgroup-objectoptions"
```
Comment (britt6612) [inline aries-site/src/examples/components/checkboxgroup/CheckBoxGroupObjectExample.js:23]: They shouldn't all be checked when you click on one ?

Comment (britt6612) [inline aries-site/src/examples/components/checkboxgroup/CheckBoxGroupObjectExample.js:23]: `valueKey="val"`

**Comment by britt6612** [comment]
```
    valueKey: PropTypes.string.description(
      `When the options array contains objects, this property indicates how
        to determine the value of each option. If a string is provided, 
        it is used as the key to retrieve each option's value.`,
    ),
  };
```

From Grommet docs then I think it should be good to go!

**Comment by taysea** [comment]
> ```
>     valueKey: PropTypes.string.description(
>       `When the options array contains objects, this property indicates how
>         to determine the value of each option. If a string is provided, 
>         it is used as the key to retrieve each option's value.`,
>     ),
>   };
> ```
> 
> From Grommet docs then I think it should be good to go!

Thank you!

File: aries-site/src/examples/components/checkboxgroup/CheckBoxGroupObjectExample.js (hunk 1 above)
Comment (taysea) [reply aries-site/src/examples/components/checkboxgroup/CheckBoxGroupObjectExample.js:23]: Good catch, thanks! added it in now.

**Pull request merged**